data-prepper-plugins/opensearch/src/test/java/com/amazon/dataprepper/plugins/sink/opensearch/OpenSearchIntegrationHelper.java

```diff
@@ -0,0 +1,254 @@
+                .map("true"::equalsIgnoreCase).orElse(false);
+        if (osFlag) {
+            // currently only external cluster is supported for security enabled testing
+            if (!Optional.ofNullable(System.getProperty("tests.rest.cluster")).isPresent()) {
+                throw new RuntimeException("cluster url should be provided for security enabled OpenSearch testing");
+            }
+        }
+
+        return osFlag;
+    }
+
+    private static String getProtocol() {
+        return isOSBundle() ? "https" : "http";
+    }
+
+
+    private static void configureHttpsClient(final RestClientBuilder builder, final Settings settings) throws IOException {
+        final Map<String, String> headers = ThreadContext.buildDefaultHeaders(settings);
+        final Header[] defaultHeaders = new Header[headers.size()];
+        int i = 0;
+        for (Map.Entry<String, String> entry : headers.entrySet()) {
+            defaultHeaders[i++] = new BasicHeader(entry.getKey(), entry.getValue());
+        }
+        builder.setDefaultHeaders(defaultHeaders);
+        builder.setHttpClientConfigCallback(httpClientBuilder -> {
+            final String userName = Optional
+                    .ofNullable(System.getProperty("user"))
+                    .orElseThrow(() -> new RuntimeException("user name is missing"));
+            final String password = Optional
+                    .ofNullable(System.getProperty("password"))
+                    .orElseThrow(() -> new RuntimeException("password is missing"));
+            final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+            credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(userName, password));
+            try {
+                return httpClientBuilder
+                        .setDefaultCredentialsProvider(credentialsProvider)
+                        // disable the certificate since our testing cluster just uses the default security configuration
+                        .setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE)
+                        .setSSLContext(SSLContextBuilder.create().loadTrustMaterial(null, (chains, authType) -> true).build());
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+
+        final String socketTimeoutString = settings.get(CLIENT_SOCKET_TIMEOUT);
+        final TimeValue socketTimeout = TimeValue
+                .parseTimeValue(socketTimeoutString == null ? "60s" : socketTimeoutString, CLIENT_SOCKET_TIMEOUT);
+        builder.setRequestConfigCallback(conf -> conf.setSocketTimeout(Math.toIntExact(socketTimeout.getMillis())));
+        if (settings.hasValue(CLIENT_PATH_PREFIX)) {
+            builder.setPathPrefix(settings.get(CLIENT_PATH_PREFIX));
+        }
+    }
+
+    private static void configureClient(RestClientBuilder builder, Settings settings) throws IOException {
```

Comment:
Is this the only copied piece of code we are leveraging? Let's call this out for the maintainers.

Reply:
Yes, I think that the best solution would be to refactor the OpenSearch tests. This would likely be a significant change and will take some time to implement. I'd like to fix these tests in Data Prepper to support OpenSearch 2.0. As a follow-up we can best determine how to best refactor the tests for OpenSearch.

Reply:
I'd also like to see if we can improve our tests - maybe we will need fewer of these components. But, I'd like to do this in a more phased approach. This phase is currently removing the inheritance model which is imposing OpenSearch internal requirements.

Reply:
I just removed the `configureClient` in favor of always using HTTPS.